pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -341,31 +407,27 @@ func (e *TokensController) createSecret(serviceAccount *api.ServiceAccount) erro
 
 	liveServiceAccount.Secrets = append(liveServiceAccount.Secrets, api.ObjectReference{Name: secret.Name})
 
-	_, err = serviceAccounts.Update(liveServiceAccount)
-	if err != nil {
+	if _, err = serviceAccounts.Update(liveServiceAccount); err != nil {
 		// we weren't able to use the token, try to clean it up.
-		glog.V(2).Infof("Deleting secret %s/%s because reference couldn't be added (%v)", secret.Namespace, secret.Name, err)
+		glog.V(2).Infof("deleting secret %s/%s because reference couldn't be added (%v)", secret.Namespace, secret.Name, err)
+
 		if err := e.client.Core().Secrets(secret.Namespace).Delete(secret.Name, nil); err != nil {
```

Comment:
It will get cleaned up if the service account goes away, but not otherwise. I'm not sure how concerned about this particular clean up I am